src/gt4py/backend/dawn_backends.py

```diff
@@ -238,41 +383,43 @@ def generate(
         definition_ir: gt_ir.StencilDefinition,
         definition_func: types.FunctionType,
         options: gt_definitions.BuildOptions,
+        **kwargs,
     ):
-        # TODO: move this import to the top and find a better way to avoid circular imports
         from gt4py import gt_src_manager
 
         cls._check_options(options)
 
         # Generate the Python binary extension (checking if GridTools sources are installed)
         if not gt_src_manager.has_gt_sources() and not gt_src_manager.install_gt_sources():
             raise RuntimeError("Missing GridTools sources.")
-
-        module_kwargs = {}
         pyext_module_name, pyext_file_path = cls.generate_extension(
-            stencil_id, definition_ir, options, module_kwargs=module_kwargs
+            stencil_id, definition_ir, options, **kwargs
         )
 
-        # Generate and return the Python wrapper class
+        if not options._impl_opts.get("disable-code-generation", False):
```

Comment:
I know the hierarchy of classes and methods is convoluted and it is not at all well documented, but the idea here is that you external backends that have their own analysis toolchain can pass a `module_info` dictionary with the information required to build the python module instead of running the internal analysis pipeline to generate the `implementation_ir`. Look at the `__call__` method of the `BaseModuleGenerator` in `base.py` for more info of how this `module_info` will be used.

Reply:
I will work on this.

Reply:
The code has been refactored to generate a `module_info` dictionary rather than generating an `implementation_ir` using the `gt4py` pipeline.